ibis/formats/parser.py

```diff
@@ -0,0 +1,213 @@
+    DataType.Type.LONGTEXT: dt.string,
+    DataType.Type.MEDIUMBLOB: dt.binary,
+    DataType.Type.MEDIUMTEXT: dt.string,
+    DataType.Type.MONEY: dt.int64,
+    DataType.Type.NCHAR: dt.string,
+    DataType.Type.NULL: dt.null,
+    DataType.Type.NVARCHAR: dt.string,
+    DataType.Type.OBJECT: dt.Map(dt.string, dt.json),
+    DataType.Type.SMALLINT: dt.int16,
+    DataType.Type.SMALLMONEY: dt.int32,
+    DataType.Type.TEXT: dt.string,
+    DataType.Type.TIME: dt.time,
+    DataType.Type.TIMETZ: dt.time,
+    DataType.Type.TINYINT: dt.int8,
+    DataType.Type.UBIGINT: dt.uint64,
+    DataType.Type.UINT: dt.uint32,
+    DataType.Type.USMALLINT: dt.uint16,
+    DataType.Type.UTINYINT: dt.uint8,
+    DataType.Type.UUID: dt.uuid,
+    DataType.Type.VARBINARY: dt.binary,
+    DataType.Type.VARCHAR: dt.string,
+    DataType.Type.VARIANT: dt.json,
+    DataType.Type.UNIQUEIDENTIFIER: dt.uuid,
+    #############################
+    # Unsupported sqlglot types #
+    #############################
+    # BIGSERIAL = auto()
+    # DATETIME64 = auto() # clickhouse
+    # ENUM = auto()
+    # INT4RANGE = auto()
+    # INT4MULTIRANGE = auto()
+    # INT8RANGE = auto()
+    # INT8MULTIRANGE = auto()
+    # NUMRANGE = auto()
+    # NUMMULTIRANGE = auto()
+    # TSRANGE = auto()
+    # TSMULTIRANGE = auto()
+    # TSTZRANGE = auto()
+    # TSTZMULTIRANGE = auto()
+    # DATERANGE = auto()
+    # DATEMULTIRANGE = auto()
+    # HLLSKETCH = auto()
+    # IMAGE = auto()
+    # IPPREFIX = auto()
+    # ROWVERSION = auto()
+    # SERIAL = auto()
+    # SET = auto()
+    # SMALLSERIAL = auto()
+    # SUPER = auto()
+    # TIMESTAMPLTZ = auto()
+    # UNKNOWN = auto()  # Sentinel value, useful for type annotation
+    # UINT128 = auto()
+    # UINT256 = auto()
+    # USERDEFINED = "USER-DEFINED"
+    # XML = auto()
+}
+
+
+class TypeParser(Singleton, Slotted):
+    @property
+    @abc.abstractmethod
+    def dialect(self) -> str:
+        """The dialect this parser is for."""
+
+    default_nullable = True
+    """Default nullability when not specified."""
+
+    default_decimal_precision: int | None = None
+    """Default decimal precision when not specified."""
+
+    default_decimal_scale: int | None = None
+    """Default decimal scale when not specified."""
+
+    default_temporal_scale: int | None = None
+    """Default temporal scale when not specified."""
+
+    default_interval_precision: str | None = None
+    """Default interval precision when not specified."""
+
+    fallback: Mapping[str, dt.DataType] = FrozenDict()
+    """Default fallback mapping of SQL string types to ibis types."""
+
+    default_fallback: Mapping[str, dt.DataType] = FrozenDict(
+        {
+            "INTERVAL YEAR TO MONTH": dt.Interval("M"),
+            "INTERVAL DAY TO SECOND": dt.Interval("ms"),
+        }
+    )
+    """Mapping of SQL string types to ibis types shared by all parser child classes."""
+
+    @property
+    def _fallback(self) -> Mapping[str, dt.DataType]:
```

Comment:
Is this used anywhere else then the following `parse()` method?

Reply:
No. I guess I could inline it.

Reply:
Done!